control-plane-operator/controllers/hostedcontrolplane/kas/kubeconfig.go

```diff
@@ -20,10 +21,17 @@ func ReconcileServiceKubeconfigSecret(secret, cert *corev1.Secret, ca *corev1.Co
 	return pki.ReconcileKubeConfig(secret, cert, ca, svcURL, "", "service", ownerRef)
 }
 
-func ReconcileServiceCAPIKubeconfigSecret(secret, cert *corev1.Secret, ca *corev1.ConfigMap, ownerRef config.OwnerRef, apiServerPort int32) error {
+func ReconcileServiceCAPIKubeconfigSecret(secret, cert *corev1.Secret, ca *corev1.ConfigMap, ownerRef config.OwnerRef, apiServerPort int32, capiClusterName string) error {
 	svcURL := InClusterKASURL(secret.Namespace, apiServerPort)
 	// The client used by CAPI machine controller expects the kubeconfig to have this key
 	// https://github.com/kubernetes-sigs/cluster-api/blob/5c85a0a01ee44ecf7c8a3c3fdc867a88af87d73c/util/secret/secret.go#L29-L33
+	// and to be labeled with cluster.x-k8s.io/cluster-name=<clusterName> so the secret can be cached by the client.
+	// https://github.com/sbueringer/cluster-api/blob/49e7f681aacd12bc467080dfcf2e405bdee3237f/main.go#L300
```

Comment:
nit: link points to individual repo, can we link to capi one?

Reply:
sure, nice catch